doc/reference/stdio.md

```diff
@@ -0,0 +1,843 @@
+
+The default buffer size is 32KB.
+
+
+#### BufferedReader-peek-u8
+```scheme
+(BufferedReader-peek-u8 buf) -> u8 or #!eof
+  buf := BufferedReader
+```
+
+Peeks the next byt from the buffer, filling it if it is empty.
+Returns `#!eof` if the end of input is reached.
+
+#### BufferedReader-read-u8
+```scheme
+(BufferedReader-read-u8 buf) -> u8 or #!eof
+  buf := BufferedReader
+```
+
+Reads a byte from the buffer.
+Returns `#!eof` if the end of input is reached.
+
+#### BufferedReader-read-u8!
+```scheme
+(BufferedReader-read-u8! buf) -> u8
+  buf := BufferedReader
+```
+
+Like `BufferedReader-read-u8`, but raises an error if the end of input is reached.
+
+#### BufferedReader-read-u16
+```scheme
+(BufferedReader-read-u16 buf) -> integer
+  buf := BufferedReader
+```
+
+Reads an usigned 16 bit integer, encoded in network byte order.
+Raises an error if the end of input is reached.
+
+#### BufferedReader-read-s16
+```scheme
+(BufferedReader-read-s16 buf) -> integer
+  buf := BufferedReader
+```
+
+Reads a signed 16 bit integer, encoded in network byte order.
+Raises an error if the end of input is reached.
+
+#### BufferedReader-read-u32
+```scheme
+(BufferedReader-read-u32 buf) -> integer
+  buf := BufferedReader
+```
+
+Reads an usigned 32 bit integer, encoded in network byte order.
+Raises an error if the end of input is reached.
+
+#### BufferedReader-read-s32
+```scheme
+(BufferedReader-read-s32 buf) -> integer
+  buf := BufferedReader
+```
+
+Reads a signed 32 bit integer, encoded in network byte order.
+Raises an error if the end of input is reached.
+
+
+#### BufferedReader-read-u64
+```scheme
+(BufferedReader-read-u64 buf) -> integer
+  buf := BufferedReader
+```
+
+Reads an unsigned 64 bit integer, encoded in network byte order.
+Raises an error if the end of input is reached.
+
+
+#### BufferedReader-read-s64
+```scheme
+(BufferedReader-read-s64 buf) -> integer
+  buf := BufferedReader
+```
+
+Reads signed 64 bit integer, encoded in network byte order.
+Raises an error if the end of input is reached.
+
+
+#### BufferedReader-read-varuint
+```scheme
+(BufferedReader-read-varuint buf (max-bits 64)) -> integer
+  buf      := BufferedReader
+  max-bits := fixnum
+```
+Reads an unsigned integer, with variadic encoding and a maximum integer length of `max-bits`
+Raises an error if the end of input is reached.
+
+#### BufferedReader-read-varint
+```scheme
+(BufferedReader-read-varint reader (max-bits 64)) -> integer
+```
+Reads a signed integer, with variadic encoding and a maximum integer length of `max-bits`
+Raises an error if the end of input is reached.
+
+#### BufferedReader-read-char
+```scheme
+(BufferedReader-read-char buf) -> char or #!eof
+  buf := BufferedReader
+```
+
+Reads a utf8-encoded character.
+Returns `#!eof` if the endo if input is reached before reading the first byte.
+Raises an error if the end of input is reached for multibyte encoded characters.
+
```

Comment:
Also an error if the next few bytes do not encode a valid utf8 character, innit?

Reply:
yes.

Reply:
actually no, it reads one byte and returns the utf8 replacement char; added specification of this behavior.